atom/browser/atom_permission_manager.cc

```diff
@@ -242,4 +242,12 @@ bool AtomPermissionManager::CheckPermissionWithDetails(
                             *details);
 }
 
+blink::mojom::PermissionStatus
+AtomPermissionManager::GetPermissionStatusForFrame(
+    content::PermissionType permission,
+    content::RenderFrameHost* render_frame_host,
+    const GURL& requesting_origin) {
+  return blink::mojom::PermissionStatus::GRANTED;
```

Comment:
Do we need to call the `check_handler_` here? If not, why not?

Reply:
^ @MarshallOfSound , @deepak1556 ?

Reply:
@MarshallOfSound Since you made both changes (adding `GetPermissionStatusForFrame` and `CheckPermissionWithDetails`) can you please fix logic here after this PR is merged?

Reply:
Oh yeah, I must have added `GetPermissionStatusForFrame` before `CheckPermissionWithDetails` landed in master.  It looks like I just took the approach that chromium took and basically replicate the `GetPermissionStatus` impl.

https://github.com/electron/electron/blob/54ad38526095f4a415ad75a1757feae4336c9db4/atom/browser/atom_permission_manager.cc#L213

I'll raise a thing to look / clean this up.

Reply:
Yes but we are returning the same value as our `GetPermissionStatus` here, there is no need for change in the impl. We return granted permission status which is used by sync javascript apis to check for permission status. The actual requesting of status happens through a different code path. But we have been discussing about changing this default value from granted to something more secure, but that's outside the scope of this PR. But it would be good to discuss and change it once this is merged.

Reply:
@deepak1556 I raised https://github.com/electron/electron/issues/14544 with this information 👍 I recently added sync permission checking support to our session API's so we could in theory wire these up 👍